src/ixdat/plotters/ms_plotter.py

```diff
@@ -397,6 +478,203 @@ def _parse_overloaded_inputs(
+                mol_lists=mol_lists,
+                unit=unit,
+                logplot=logplot,
+                legend=legend,
+                **kwargs,
+            )
+
+        measurement.spectrum_series.heat_plot(
+            ax=axes[ms_spec_axes],
+            tspan=tspan,
+            xspan=xspan,
+            cmap_name=cmap_name,
+            make_colorbar=make_colorbar,
+            max_threshold=max_threshold,
+            min_threshold=min_threshold,
+            scanning_mask=scanning_mask,
+        )
+
+        return axes
+
+    def plot_measurement_vs(
+        self,
+        *,
+        x_name,
+        measurement=None,
+        axes=None,
+        mass_list=None,
+        mass_lists=None,
+        mol_list=None,
+        mol_lists=None,
+        tspan=None,
+        tspan_bg=None,
+        remove_background=None,
+        unit=None,
+        logplot=True,
+        legend=True,
+        xspan=None,
+        cmap_name="inferno",
+        make_colorbar=False,
+        emphasis="top",
+        ms_data="top",
+        max_threshold=None,
+        min_threshold=None,
+        scanning_mask=None,
+        _sort_indicies=None,
+        **kwargs,
+    ):
+
+        if logplot is None:
+            logplot = not mol_lists and not mass_lists
+
+        if not axes:
+            if ms_data == "top":
+                n_bottom = 1
+                n_top = 2 if (mass_lists or mol_lists) else 1
+                ms_axes = 0
+                ms_spec_axes = 1
+            else:
+                n_top = 1
+                n_bottom = 2 if (mass_lists or mol_lists) else 1
+                ms_axes = 1
+                ms_spec_axes = 0
+
+            axes = self.new_two_panel_axes(
+                n_bottom=n_bottom,
+                n_top=n_top,
+                emphasis=emphasis,
+            )
+
+        measurement = measurement or self.measurement
+
+        if (
+            mass_list
+            or mass_lists
+            or mol_list
+            or mol_lists
+            or hasattr(measurement, "mass_list")
+        ):
+            # then we have MS data!
+            self.ms_plotter.plot_vs(
+                x_name=x_name,
+                measurement=measurement,
+                axes=[axes[ms_axes], axes[2]]
+                if (mass_lists or mol_lists)
+                else [axes[ms_axes]],
+                tspan=tspan,
+                tspan_bg=tspan_bg,
+                remove_background=remove_background,
+                mass_list=mass_list,
+                mass_lists=mass_lists,
+                mol_list=mol_list,
+                mol_lists=mol_lists,
+                unit=unit,
+                logplot=logplot,
+                legend=legend,
+                **kwargs,
+            )
+
+        _tseries = measurement.spectrum_series.field.axes_series[0]
+        _v = measurement.grab_for_t(item=x_name, t=_tseries.t)
+
+        if not _sort_indicies:
+            _sort_indicies = np.argsort(_v)
```

Comment:
Exactly. I agree, and could not figure out the best way to avoid funky xaxis.  

heat_plot() takes the default xspan[0], xspan[-1]. 

So scanning up and down yields a x axis from 34 degC to 53 degC. Even though the temperature in the middle is 250 degC.

By sorting this way you always get data and xaxis to match from lowest to highest value. 

Alternatively, the xaxis have to be manipulated more directly with the entire list of x_name values .